app/frontend/src/services/adminService.js

```diff
@@ -6,6 +6,22 @@ export default {
   // Form calls
   //
 
+  /**
+   * @function addFormUser
+   * Add a form user role (specific administrative task, for real form/user/role management see rbac service)
+   * @param {formId} formId The request body for the relationships
+   * @param {userId} userId The request body for the relationships
+   * @param {String[]} [roles] The list of roles to add
+   * @returns {Promise} An axios response
+   */
+  addFormUser(userId, formId, roles) {
+    return appAxios().put(`${ApiRoutes.ADMIN}${ApiRoutes.FORMS}/${formId}/addUser`, roles.map(role => ({
+      userId: userId,
+      formId: formId,
+      role: role
+    })), { params: { userId: userId } });
```

Comment:
For that new CHEFS endpoint, if the formId is required via path, and the query parameter is requiring a userId, then the body should just be an array of roles instead of the current array of formId, role and userId tuples.

Reply:
Discovered this limitation/behavior with query param being required came from the previous Objection 2.x to 3.x migration where querying with an undefined array would cause Objection to error out in the newer version. Decided for now to leave this the way its implemented in this PR and to update spec as needed to reflect current implemented behavior. Will revisit this at a later time.